pandas/core/arrays/arrow/array.py

```diff
@@ -726,6 +727,13 @@ def _evaluate_op_method(self, other, op, arrow_funcs):
         result = pc_func(self._pa_array, other)
         return type(self)(result)
 
+    @staticmethod
+    def _evaluate_binary_repeat(binary, integral):
+        if not pa.types.is_integer(integral.type):
+            raise TypeError("Can only string multiply by an integer.")
+        pa_integral = pc.if_else(pc.less(integral, 0), 0, integral)
+        return pc.binary_repeat(binary, pa_integral)
```

Comment:
Can you inline these where needed. I don't think a whole new function is needed for it

Reply:
@mroeschke, can do, but there's 2 callers of it, so if we remove the function, the logic for calculating the repeat count to pass to binary_repeat would need to be repeated in both callers. Additionally, I believe the helper function simplifies the code because there's 2 scenarios:

1. binary is lhs and int is rhs
2. binary is rhs and int is lhs

When we call binary_repeat, we need to always ensure the binary is the left arg and the integral is the right arg, imo the helper function simplifies the code and makes it a bit more readable. Let me know if you would still prefer logic of _evaluate_binary_repeat to be inlined, or if I misunderstood your suggestion.

Reply:
I would still prefer to inline this for now. We don't really use staticmethods in the codebase

Reply:
I see, inlined.